stix_shifter_modules/cybereason/stix_transmission/api_client.py

```diff
@@ -0,0 +1,68 @@
+from stix_shifter_utils.stix_transmission.utils.RestApiClient import RestApiClient
+from stix_shifter_utils.utils import logger
+
+DEFAULT_LIMIT = 10000
+DEFAULT_OFFSET = 0
+
+
+class APIClient:
+    LOGIN_ENDPOINT = 'login.html'
+    QUERY_ENDPOINT = "rest/visualsearch/query/simple"
+    LOGOFF_ENDPOINT = 'logout'
+
+    def __init__(self, connection, configuration):
+        self.logger = logger.set_logger(__name__)
+        headers = {'Content-Type': 'application/json'}
+        url_modifier_function = None
+        self.auth = configuration.get('auth')
+        self.client = RestApiClient(connection.get('host'),
+                                    connection.get('port', None),
+                                    headers,
+                                    url_modifier_function=url_modifier_function,
+                                    cert_verify=connection.get('selfSignedCert', True),
+                                    sni=connection.get('sni', None)
+                                    )
+        self.timeout = connection['options'].get('timeout')
+
+    def ping_box(self):
+        """
+        Ping the Data Source
+        :return: Response object
+        """
+        headers = {}
+
+        query = '{"queryPath": [{"requestedType": "Connection", "filters": [], "isResult": true}],' \
+                '"totalResultLimit": 1, "perGroupLimit": 1,"templateContext": "SPECIFIC"}'
+        headers['Cookie'] = self.get_cookies_id()
+        return self.client.call_api(self.QUERY_ENDPOINT, 'POST', headers=headers, data=query)
```

Comment:
is getting the cookie id not enough to determine whether the datasource is reachable? not sure if it is required to make a query for the ping in this case.

Reply:
The cookie id which is received from login session is required to get results from the data source. Getting cookie id alone will not ensure that data source is reachable and ping(responding to native data source query) is successful.